failsafe/retry_policy.py

```diff
@@ -10,14 +10,46 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from datetime import timedelta
+import random
+
+
+class Backoff:
+    def __init__(self, delay, max_delay, factor=2, jitter=False):
+        assert isinstance(delay, timedelta), "`delay` must be an instance of `datetime.timedelta`."
+        assert isinstance(max_delay, timedelta), "`max_delay` must be an instance of `datetime.timedelta`."
+        self.delay = delay
+        self.max_delay = max_delay
+        self.factor = factor
+        self.jitter = jitter
+
+    def for_attempt(self, attempt):
+        delay = self.delay.total_seconds()
+        duration = float(delay * pow(self.factor, attempt))
+        max_delay = self.max_delay.total_seconds()
+
+        if self.jitter is True:
+            duration += random.random()
```

Comment:
Implementing jitter like this means that jitter can be significant for values under the tenth of a second, but non-significant as we go higher, say to the tens of seconds. Just by using `*=` instead of `*=` you'd make jitter significant no matter the order of magnitude.

Reply:
I've also seen `duration = random.random() * (duration - delay) + delay` used in the wild, but I'm not sure what the logic behind that math is